src/FindApprenticeshipTraining/SFA.DAS.FindApprenticeshipTraining/Application/Courses/Queries/GetCourseByLarsCode/GetCourseByLarsCodeQueryHandler.cs

```diff
@@ -0,0 +1,71 @@
+﻿using MediatR;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Extensions;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests.RoatpV2;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses.Courses;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses.RoatpV2;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.FindApprenticeshipTraining.Application.Courses.Queries.GetCourseByLarsCode;
+
+public sealed class GetCourseByLarsCodeQueryHandler(
+    ICoursesApiClient<CoursesApiConfiguration> _coursesApiClient,
+    IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> _roatpCourseManagementApiClient
+) : IRequestHandler<GetCourseByLarsCodeQuery, GetCourseByLarsCodeQueryResult>
+{
+    public const string KsbsSkillsType = "Skill";
+    public const string KsbsKnowledgeType = "Knowledge";
+    public const string KsbsBehaviorType = "Behaviour";
+
+    public async Task<GetCourseByLarsCodeQueryResult> Handle(GetCourseByLarsCodeQuery query, CancellationToken cancellationToken)
+    {
+        var coursesApiStandardResponse = await _coursesApiClient.GetWithResponseCode<StandardDetailResponse>(
+            new GetStandardDetailsByIdRequest(
+                query.LarsCode.ToString()
+            )
+        );
+
+        coursesApiStandardResponse.EnsureSuccessStatusCode();
+
+        StandardDetailResponse standardDetails = coursesApiStandardResponse.Body;
+
+        ApprenticeshipFunding apprenticeshipFunding = standardDetails.ApprenticeshipFunding[standardDetails.ApprenticeshipFunding.Count - 1];
+
+        var courseTrainingProvidersCountResponse =
+            await _roatpCourseManagementApiClient.GetWithResponseCode<GetCourseTrainingProvidersCountResponse>(
+                new GetCourseTrainingProvidersCountRequest(
+                    [query.LarsCode],
+                    query.Distance,
+                    query.Lat,
+                    query.Lon
+                )
+        );
+
+        courseTrainingProvidersCountResponse.EnsureSuccessStatusCode();
+
+        CourseTrainingProviderCountModel trainingCourseCountDetails = 
+            courseTrainingProvidersCountResponse.Body.Courses.Count > 0 ?
+                courseTrainingProvidersCountResponse.Body.Courses[0] : 
+                null;
```

Comment:
If the inner returns ok response, the count should be given even if it is zero, should never be null

Reply:
I see the inner is returning a list of courses.